core_backend/app/llm_call/dashboard.py

```diff
@@ -40,13 +48,37 @@ async def generate_topic_label(
     user_id: int,
     context: str,
     sample_texts: list[str],
+    topic_model: BERTopic,
 ) -> dict[str, str]:
     """
-    Generates topic labels for example queries
+    Generates topic labels for example queries.
     """
     if topic_id == -1:
         return {"topic_title": "Unclassified", "topic_summary": "Not available."}
 
+    if DISABLE_DASHBOARD_LLM:
+        logger.info("LLM functionality is disabled. Generating labels using KeyBERT.")
+        # Use KeyBERT-inspired method to generate topic labels
+        # Assume topic_model is provided
+        topic_info = topic_model.get_topic(topic_id)
+        if not topic_info:
+            logger.warning(f"No topic info found for topic_id {topic_id}.")
+            return {"topic_title": "Unknown", "topic_summary": "Not available."}
+
+        # Extract top keywords
+        top_keywords = [word for word, _ in topic_info]
+        topic_title = ", ".join(top_keywords[:3])  # Use top 3 keywords as title
+        # Use all keywords as summary
+        # Line formatting looks odd since 'pre-wrap' is enabled on the frontend
+        topic_summary = f"""{" ".join(top_keywords)}
+
+Hint: To enable full AI summaries please set the DASHBOARD_LLM environment variable to "True" in your configuration."""  # noqa: E501
+        logger.info(
+            f"Generated topic label for topic_id {topic_id} using basic keywords."
+        )
+        return {"topic_title": topic_title, "topic_summary": topic_summary}
```

Comment:
What do you think about using a Schema instead of a dict?

Reply:
My only counterargument would be that we do slot both of these into a schema later on [here](https://github.com/IDinsight/ask-a-question/blob/e03110d54ea7a9d90bae41208fff3be76e74920a/core_backend/app/dashboard/schemas.py#L180) so it might overkill to create a smaller schema for just this dict + then have Topics include that smaller schema within it.